### PR TITLE
MAINT: have `qn_domains`-keys in Node/EdgeSettings be typed

### LINKED ISSUE #281: Specify key type of `EdgeSettings.qn_domains` and `NodeSettings.qn_domains`
The key of `qn_domains` in `NodeSettings` and `EdgeSettings` can be typed with an alias like [`EdgeQuantumNumber`](https://github.com/ComPWA/qrules/blob/e60ff895b4cf84be18291b6e4fe0ab516e75a493/src/qrules/quantum_numbers.py#L111-L131) instead of `Any`, see https://github.com/ComPWA/qrules/pull/278#discussion_r1710182559). So the type would be something like:

```python
EdgeQuantumNumberType = Union[
    type[EdgeQuantumNumbers.pid],
    type[EdgeQuantumNumbers.mass],
    type[EdgeQuantumNumbers.width],
    ...
    type[EdgeQuantumNumbers.g_parity],
]
NodeQuantumNumberType = Union[
    type[NodeQuantumNumbers.l_magnitude],
    ...
]
```

### PR DESCRIPTION
**Pull request by grayson-helmholz**
In order to have more reliable type-checking the keys of the `qn_domains`-dictionary should be the types of the respective domains themselves. This avoids for example the use of wrong quantum numbers (EdgeQNs for a node, etc.)

Closes #281

### CODE REVIEW

**Comment by grayson-helmholz** [comment]
Hopefully merging of #289 will make it easier to have "CI / doc / nb" pass

File: docs/conf.py [hunk 1]
```diff
@@ -259,6 +262,13 @@
 nb_execution_show_tb = True
 nb_execution_timeout = -1
 nb_output_stderr = "remove"
+nitpick_temp_eqdge_names = list(attrs.fields_dict(EdgeQuantumNumbers))
+nitpick_temp_node_names = list(attrs.fields_dict(NodeQuantumNumbers))
+nitpick_temp_names = [*nitpick_temp_eqdge_names, *nitpick_temp_node_names]
```
Comment (redeboer) [inline docs/conf.py:267]:
```suggestion
nitpick_temp_names = [
    *attrs.fields_dict(EdgeQuantumNumbers),
    *attrs.fields_dict(NodeQuantumNumbers),
]
```

**Review by redeboer — CHANGES_REQUESTED** [review]

File: docs/conf.py [hunk 2]
```diff
@@ -259,6 +262,23 @@
...
+    return [
+        attr
+        for attr in dir(some_type)
+        if type(getattr(some_type, attr)) is typing.NewType
+    ]
+
+
+nitpick_temp_edge_names = pick_newtype_attrs(EdgeQuantumNumbers)
+nitpick_temp_node_names = pick_newtype_attrs(NodeQuantumNumbers)
+nitpick_temp_names = [*nitpick_temp_edge_names, *nitpick_temp_node_names]
```
Comment (redeboer) [inline docs/conf.py:277]:
```suggestion
nitpick_temp_names = [
    *pick_newtype_attrs(EdgeQuantumNumbers),
    *pick_newtype_attrs(NodeQuantumNumbers),
]
```

File: docs/conf.py [hunk 3]
```diff
@@ -259,6 +262,23 @@
...
+
+
+def pick_newtype_attrs(some_type: type) -> list:
+    return [
+        attr
+        for attr in dir(some_type)
+        if type(getattr(some_type, attr)) is typing.NewType
+    ]
+
+
```
Comment (redeboer) [inline docs/conf.py:274]: Could you move this one before all the variables (options) in this file? So right after the imports.

**Review by redeboer — CHANGES_REQUESTED** [review]

File: docs/conf.py [hunk 4]
```diff
@@ -259,6 +271,18 @@
 nb_execution_show_tb = True
 nb_execution_timeout = -1
 nb_output_stderr = "remove"
+
+
+nitpick_temp_edge_names = pick_newtype_attrs(EdgeQuantumNumbers)
+nitpick_temp_node_names = pick_newtype_attrs(NodeQuantumNumbers)
```
Comment (redeboer) [inline docs/conf.py:277]: These are duplicate now
```suggestion
```

**Review by redeboer — CHANGES_REQUESTED** [review]

**Pull request merged**